core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -133,6 +133,8 @@ public MatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
                     } else {
                         throw new ParsingException(parser.getTokenLocation(), "[match] query does not support [" + currentFieldName + "]");
                     }
+                } else {
+                    throw new ParsingException(parser.getTokenLocation(), "[match] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?